tinygrad/lazy.py

```diff
@@ -210,6 +210,12 @@ def shuffle_and_prune_movement_ops(self, st: ShapeTracker, op: MovementOps, arg:
   def reduce_op(self:LazyBuffer, op:ReduceOps, new_shape:Tuple[int, ...]) -> LazyBuffer:
     if self.shape == tuple(new_shape): return self
     srcs = _push_movement_ops((self,)) if SHUFFLE_MOVEMENT_OPS else (self,)
+
+    # Split multi-axis reduction into two kernels on Metal for perf
+    if sum(shape_differences := [new != old for new, old in zip(new_shape, self.shape)]) > 1 and self.device == "METAL":
```

Comment:
Should probably swap the order on these to reduce the hit for non-metal backends, assuming we keep the check.

fwiw I saw a big speedup on cuda on the first iteration of this PR